bindings/rust/bench/src/bin/graph_perf.rs

```diff
@@ -0,0 +1,159 @@
+        SVGBackend,
+    },
+    series::LineSeries,
+    style::{AsRelative, Color, IntoFont, Palette, Palette99, RGBAColor, BLACK, WHITE},
+};
+use semver::Version;
+use std::{collections::HashMap, error::Error};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let mut csv_reader = Reader::from_path("historical-perf/perf.csv")?;
+
+    // map bench name -> points (x, y, y_bar)
+    let mut map: HashMap<&str, Vec<(Version, f64, f64)>> = HashMap::new();
+
+    // get headers of csv as Vec<String>
+    let headers = csv_reader.headers()?.clone();
+    let headers: Vec<String> = headers
+        .iter()
+        .skip(1) // skip first header "tag"
+        .step_by(2) // skip stderr headers
+        .map(|s| s.to_string())
+        .collect::<Vec<String>>();
+
+    // initialize data vectors
+    for header in headers.iter() {
+        map.insert(header, Vec::new());
+    }
+
+    // read in all the tags as Versions
+    let mut tag_names = Reader::from_path("historical-perf/perf.csv")?
+        .records()
+        .map(|record_res| Version::parse(&record_res.unwrap()[0][1..]).unwrap())
+        .collect::<Vec<_>>();
+    let num_tags = tag_names.len();
+
+    // fill in missing versions (1.3.15, 1.3.30-1.3.37)
+    tag_names.push(Version::new(1, 3, 15));
+    tag_names.extend((30..38).map(|p| Version::new(1, 3, p)));
+    tag_names.sort();
+
+    // get the indices of all of the tags for plotting
+    let tag_names_to_index = tag_names
+        .iter()
+        .enumerate()
+        .map(|(i, version)| (version, i))
+        .collect::<HashMap<_, _>>();
+
+    // keep track of ymax for plotting range
+    let mut y_max = 0.0;
+
+    // read in all the data
+    // loop through each record (row in csv), one per tag/version
+    for record in csv_reader.records().map(|res| res.unwrap()) {
+        let mut record_iter = record.into_iter();
+
+        // first item in record is tag
+        let tag = Version::parse(&record_iter.next().unwrap()[1..])?;
+
+        // every two items after is y/y-stderr pair
+        // loop through each header and track (tag_name, y, y_bar)
+        for header in headers.iter() {
+            let (y, y_stderr) = (
+                record_iter.next().unwrap().parse()?,
+                record_iter.next().unwrap().parse::<f64>()?,
+            );
+            if y > y_max {
+                y_max = y;
+            }
+            map.get_mut(header.as_str())
+                .unwrap()
+                .push((tag.clone(), y, y_stderr * 1.96)); // 1.96 is stderr coefficient for 95% confidence interval
+        }
+    }
+
+    // setup plotting
+    let drawing_area =
+        SVGBackend::new("historical-perf/historical-perf.svg", (1000, 500)).into_drawing_area();
+    drawing_area.fill(&WHITE)?;
+
+    let mut ctx = ChartBuilder::on(&drawing_area)
+        .caption(
+            "Handshake performance over versions since Jun 2022",
```

Comment:
v1.3.16 being the last tag benched is sort of hard coded everywhere since before v1.3.16 all of the rust bindings were in `s2n_tls::raw::*` instead of just `s2n_tls::*`, and it'd be a hassle to refactor the benching code to do that for us. The date is just to provide context of how much time the graph is showing, because otherwise you'd only see the version numbers.